src/Encoder.php

```diff
@@ -56,18 +57,56 @@ public function encode(string $text): array
                 continue;
             }
 
-            $piece = EncodeUtil::toBytes($match);
-            $rank = $this->vocab->tryGetRank($piece);
+            $tokenBytes = EncodeUtil::toBytes($match);
 
-            if ($rank !== null) {
```

Comment:
Can we keep the `encode` method unchanged?

Strategy of lookup in vocab (`Vocab::tryGetRank()`) is more performant than the `mergeBytePairs()`.

Reply:
Sure.